### DISCUSSION TITLE
DataList: How to get nodes with selected item(s)

### DISCUSSION THREAD

**Discussion by greystate** [leekelleher/umbraco-contentment#248] [Question] [answered]
Hi there 👋

Looking for some advice/pointers regarding the issue of selecting nodes that have certain values selected in a DataList property editor.

For example, if I've been using a DataList for selecting some "available in" countries on a product node, is there a recommended way to efficiently select the products that include a specific country in their selection? Would it make sense (or even possible) to have a built-in method to do this?

I can use various versions of the traversal methods to get at them (XPath might even work well, depending on the PropertyValueConverters) - but of course I worry about their efficiency, as there seem to always someone who knows "a better way" 😁 

/Chriztian

**Comment by leekelleher** [comment, marked as answer]
@greystate Would the Examine data-source be useful? e.g. `+availableIn: GB +availableIn: DK`

(It's currently [sorely undocumented](https://github.com/leekelleher/umbraco-contentment/blob/develop/docs/data-sources/data-source--examine.md) - but hopefully the in-datatype instructions will make sense)

**Reply by greystate** [reply]
Thanks @leekelleher — I haven't dared trying that one yet 😬 - though for this, I was thinking of how to grab the selected nodes in code (e.g. in a View)... I seem to remember nuPickers provided some trickery to ease that part 🤔

**Reply by leekelleher** [reply]
It's one of those "flexibility vs simplicity" cases, because the Examine data-source lets you set what the value of the Examine result can be, it's up to you how you deal with it on the frontend.

![Screenshot 2022-08-30 103901](https://user-images.githubusercontent.com/209066/187404603-c5209cb7-443c-4c0f-a10e-be1757226c56.png)

The default value is set to `__Key`, which is the Guid for the content node, so on the frontend, a call to something like `var content = Umbraco.Content(value);` should do it. 🤞
